src/software/ai/intent/intent.cpp

```diff
@@ -44,5 +58,30 @@ void Intent::setMotionConstraints(const std::set<MotionConstraint> &motion_const
 
 std::set<MotionConstraint> Intent::getMotionConstraints(void) const
 {
-    return motion_constraints;
+    return this->motion_constraints;
+}
+
+std::optional<NavigatorParams> Intent::getNavigatorParams() const
+{
+    return navigator_params;
+}
+
+PrimitiveMsg Intent::getPrimitiveMsg() const
+{
+    return primitive_msg;
+}
+
+PrimitiveMsg Intent::getUpdatedPrimitiveMsg(Point destination, double final_speed) const
+{
+    return primitive_msg;
+}
+
+void Intent::updateNavigatorParams(unsigned int robot_id, Point destination,
```

Comment:
I'm curious what a case would be that these navigator params can't be provided / derived directly from the primitive params? ie. is there an example of how we would use these aside form the MoveIntent?

Reply:
Eventually, we will have `MoveIntent`, `AutokickIntent`, and `AutochipIntent`, each with their namesake PrimitiveMsg internally. Each will all want to use path planning and will override NavigatorParams. However, I think we should decouple "having a particular `PrimitiveMsg`" from "I want to use the path planner". First, I don't think the Navigator should digging around PrimitiveMsgs. Second, I think PivotIntent, for example, might define a small trajectory for a MovePrimitiveMsg that it doesn't want the navigator to interfere with.

Reply:
I'll try to come up with a better way to do this

Reply:
Ok yea I get what you're going for. Happy to help brainstorm solutions, there's definiltey a few points that could introduce confusion (ie. if an intent is navigating, how does one know what the navigator is responsible for vs the primitive)

Reply:
I've tried to decouple it some more by cutting back on the intent abstraction. Please take another look